go/host/p2p/p2p.go

```diff
@@ -126,10 +128,44 @@ func (p *p2pImpl) SendBatches(batchMsg *host.BatchMsg, to string) error {
 		return fmt.Errorf("could not encode batches using RLP. Cause: %w", err)
 	}
 
-	msg := message{Type: msgTypeBatches, Contents: encodedBatchMsg}
+	msg := message{Sender: p.ourAddress, Type: msgTypeBatches, Contents: encodedBatchMsg}
 	return p.send(msg, to)
 }
 
+// Status returns the current status of the lib
+func (p *p2pImpl) Status() *host.P2PStatus {
+	return p.status.status()
+}
+
+// HealthCheck returns whether the p2p is considered healthy
+// Currently it considers itself unhealthy
+// if there's more than 100 failures on a given fail type
+// if there's a known peer for which a message hasn't been received
+func (p *p2pImpl) HealthCheck() bool {
+	const thresholdPerFailure = 100
```

Comment:
I've got no intuition for whether 100 is a good threshold - do we expect failures with our self-run validator set on testnet? Maybe make this 0 for now with a todo to revisit what the condition should be once we've got experience with monitoring it?

Reply:
100 per 10 min seems like a good alert threshold because we do seem to have occasional p2p failures but nothing systemic like a "peer being disconnected".